### ISSUE TITLE
Typings file causes a compile error.

### ISSUE THREAD

**Issue by Softdevsuperstar** [date-fns/date-fns#508]
I installed date-fns package in my typescript(v2.3.3) project and tried to use it. But when i tried to compile the project,  `typings.d.ts` file causes a compile error.
Anybody had same issue before or know the fix? The error description is as follows.
`TypeScript error: /Users/gabrielschippers/Desktop/cubik/cubik-app/shared/node_modules/date-fns/typings.d.ts(35,18): Error TS1110: Type expected.`

**Comment by kossnocorp** [comment]
Interesting, which version you use?

```bash
npm ls date-fns
# or
yarn list date-fns
```

**Comment by GabrielSchippers** [comment]
Gabriels-Mac:shared gabrielschippers$ yarn list date-fns
yarn list v0.24.6
└─ date-fns@2.0.0-alpha.1
✨  Done in 0.26s.

I have got this when run 'yarn list date-fns'.

**Comment by kossnocorp** [comment]
@GabrielSchippers I see, it's a pre-release v2 version. See change log: https://github.com/date-fns/date-fns/blob/master/CHANGELOG.md#unreleased.

In your case, the solution will be to lock at v1.

**Comment by kossnocorp** [comment]
@GabrielSchippers I see, it's a pre-release v2 version. See change log: https://github.com/date-fns/date-fns/blob/master/CHANGELOG.md#unreleased.

In your case, the solution will be to lock at v1.

**Comment by GabrielSchippers** [comment]
Thank you for your good support.

But still I have error.

I uninstalled date-fns@2.0.0-alpha.1 and installed this '"date-fns": "1.28.5"'

Gabriels-Mac:shared gabrielschippers$ yarn list date-fns
yarn list v0.24.6
└─ date-fns@1.28.5
✨  Done in 0.25s.

As you can see, version is changed already.

And the only difference in result is the error line changed.

'TypeScript error: /Users/gabrielschippers/Desktop/cubik/cubik-app/shared/node_modules/date-fns/typings.d.ts(**674,26**): Error TS1110: Type expected.'

Any more suggestions? Please.

**Comment by leshakoss** [comment]
For reference, here are the lines which cause the error:
v1: https://github.com/date-fns/date-fns/blob/v1/typings.d.ts#L674
v2: https://github.com/date-fns/date-fns/blob/master/typings.d.ts#L35

**Comment by leshakoss** [comment]
@GabrielSchippers I've managed to get the same error by downgrading TypeScript to `^1.8.0` but it's gone in `^2.0.0`. Could you please double check which version of TS do you use? Thanks

**Comment by leshakoss** [comment]
@GabrielSchippers and also could you please run the date-fns' TypeScript smoke test on your configuration:
```sh
git clone https://github.com/date-fns/date-fns.git
cd date-fns/examples/typescript
npm install
npm run build
```
If it works, could you please provide us with the exact line which causes the error? Thanks in advance

**Comment by GabrielSchippers** [comment]
Actually I am using Typescript 2.3.4.

And I have got this error when I run `npm run build`

`Gabriels-Mac:typescript gabrielschippers$ npm run build

> date-fns-example-typescript@0.1.0 build /Users/gabrielschippers/Desktop/Test/date-fns/examples/typescript
> yarn run build-date-fns && yarn run build-typescript && yarn run build-webpack

yarn run v0.24.6
$ env PACKAGE_OUTPUT_PATH="$(pwd)/node_modules/date-fns" ../../scripts/build/package.sh 
You have mistakenly installed the `babel` package, which is a no-op in Babel 6.
Babel's CLI commands have been moved from the `babel` package to the `babel-cli` package.

    npm uninstall babel
    npm install --save-dev babel-cli

See http://babeljs.io/docs/usage/cli/ for setup instructions.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! date-fns-example-typescript@0.1.0 build: `yarn run build-date-fns && yarn run build-typescript && yarn run build-webpack`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the date-fns-example-typescript@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gabrielschippers/.npm/_logs/2017-06-21T19_38_33_875Z-debug.log
`
What's wrong?
